react/client/src/components/contexts/RoutingContext.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { withRouter, RouteComponentProps } from 'react-router-dom';
+
+interface RoutingProviderProps extends RouteComponentProps<any> {
+  children: React.ReactNode;
+}
+
+export const RoutingContext = React.createContext<any>(undefined);
+
+const RoutingContextProvider = ({
+  children,
+  history,
+}: RoutingProviderProps) => {
+  const url = history.location.pathname;
+  const pageNumber = Number(url.slice(url.indexOf('/form') + 6)) || 0;
```

Comment:
What does this +6 do?

Reply:
Since we are looking at the character index of '/form', +6 characters past that index would be everything that comes after '/form/', the current page number(s)

Maybe there is a better way to do this whole thing though.

Reply:
Oh sure. Everything else is great and we'll re-explore this particular part later.

Reply:
Thanks!